server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -332,9 +332,18 @@ public PublicationContext newPublicationContext(
         boolean isRemotePublicationEnabled,
         PersistedStateRegistry persistedStateRegistry
     ) {
+        if (isRemotePublicationEnabled == true) {
+            for (DiscoveryNode node : clusterChangedEvent.state().nodes().getNodes().values()) {
+                // if a node is non-remote then created local publication context
+                if (node.isRemoteStatePublicationEnabled() == false) {
+                    break;
+                }
+                // if all nodes are remote then create remote publication context
+                return new RemotePublicationContext(clusterChangedEvent, persistedStateRegistry);
+            }
+        }
```

Comment:
WOuldn't this for loop do only the first iteration and either `break` or  `return`

Reply:
Yes. Will move this out of the loop